### ISSUE TITLE
[feature request] In-place udpate support resources

### ISSUE THREAD

**Issue by zmberg** [openkruise/kruise#1212]
**What would you like to be added**:
The [In-place Pod Vertical Scaling feature](https://github.com/kubernetes/kubernetes/pull/102884) have been merged in k8s upstream, so openKruise in-place update need support resources, related workloads:
- CloneSet
- Advanced StatefulSet
- Advanced DaemonSet
- SidecarSet

**Comment by ABNER-1** [comment]
I encountered two errors while adding e2e test cases:

1. When modifying the image and increasing resources simultaneously, I encountered a RunContainerError. I have already submitted an issue to the Kubernetes community at https://github.com/kubernetes/kubernetes/issues/127356. 

2. When I scaled the CPU limit of resources from 100m to 200m, I observed that occasionally the status was first changed by kubelet from `100m InProcess` to 200m, and then it was set back to `100m InProcess` by kubelet, and after a while, it finally became 200m.
